spec/migrations/20180712122000_remove_host_provisioning_spec.rb

```diff
@@ -7,9 +7,9 @@
     let(:miq_approval_stub) { migration_stub :MiqApproval }
 
     it 'only removes Host Provision request instances' do
-      miq_request_stub.create!(:type => 'MiqProvisionRequest', :miq_approvals => [miq_approval_stub.create!])
-      miq_request_stub.create!(:type => 'MiqHostProvisionRequest', :miq_approvals => [miq_approval_stub.create!])
-      miq_request_stub.create!(:type => 'ServiceTemplateProvisionRequest', :miq_approvals => [miq_approval_stub.create!])
+      miq_request_stub.create!(:type => 'MiqProvisionRequest', :miq_approval_ids => [miq_approval_stub.create!.id])
+      miq_request_stub.create!(:type => 'MiqHostProvisionRequest', :miq_approval_ids => [miq_approval_stub.create!.id])
+      miq_request_stub.create!(:type => 'ServiceTemplateProvisionRequest', :miq_approval_ids => [miq_approval_stub.create!.id])
```

Comment:
avoid calling the association methods because rails does a class lookup check which will fail with rails 7 since our stub class constant has been removed by rails 7 by the time this is run and so it doesn't match the expected association class object.